modules/oe_corporate_countries_address/tests/src/Kernel/AvailableCountriesSubscriberTest.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\oe_corporate_countries_address\Kernel;
+
+use Drupal\entity_test\Entity\EntityTest;
+use Drupal\field\Entity\FieldConfig;
+use Drupal\field\Entity\FieldStorageConfig;
+use Drupal\Tests\oe_corporate_countries\Kernel\CorporateCountriesRdfKernelTestBase;
+
+/**
+ * Tests the AvailableCountriesSubscriber class.
+ */
+class AvailableCountriesSubscriberTest extends CorporateCountriesRdfKernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'user',
+    'field',
+    'entity_test',
+    'address',
+    'oe_corporate_countries',
+    'oe_corporate_countries_address',
+  ];
+
+  /**
+   * Name of the field.
+   *
+   * @var string
+   */
+  protected string $fieldName = 'field_country';
+
+  /**
+   * Name of the field type.
+   *
+   * @var string
+   */
+  protected string $fieldTypeName = 'address_country';
+
+  /**
+   * Name of the entity type.
+   *
+   * @var string
+   */
+  protected string $entityTypeName = 'entity_test';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    FieldStorageConfig::create([
+      'field_name' => $this->fieldName,
+      'entity_type' => $this->entityTypeName,
+      'type' => $this->fieldTypeName,
+    ])->save();
+
+    FieldConfig::create([
+      'label' => 'My field',
+      'field_name' => $this->fieldName,
+      'entity_type' => $this->entityTypeName,
+      'bundle' => $this->entityTypeName,
+    ])->save();
+
+  }
+
+  /**
+   * Tests the deprecated countries are removed from the countries array.
+   * */
+  public function testGetAvailableCountriesArray(): void {
+
+    // Get available_countries from field.
+    // This doesn't have the deprecated countries.
+    $entity = EntityTest::create();
+    $availableCountries = $entity->get('field_country')->appendItem()->getAvailableCountries();
+
+    // Test: Check that the format is correct and that we do not have a deprecated country.
+    // This also checks that the array is keyed correctly.
+    $this->assertEquals([ "XK" => "XK", "AX" => "AX", "BE" => "BE", "TF" => "TF", "IT" => "IT" ], $availableCountries,
+      'Available countries is correctly formatted');
+  }
```

Comment:
We should respect the 80 chars limit on comments and use single quotes in arrays.
```suggestion
    // Assert that the format is correct and there no deprecated countries.
    $expected_countries = [
      'XK' => 'XK',
      'AX' => 'AX',
      'BE' => 'BE',
      'TF' => 'TF',
      'IT' => 'IT',
    ];
    $this->assertEquals($expected_countries, $actual_countries,
      'Available countries is correctly formatted.');
```

Reply:
We did that change. Thanks.

Reply:
The `$expected_countries` is still using double quotes.